docs/testing/webdriverio/03-mocking.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Mocking
+sidebar_label: Mocking
+description: Mocking
+label: Mocking
+---
+
+# Mocking
+
+WebdriverIO has support for file based module mocking as well as mocking of entire dependencies of your project. The framework provides a set of primitives for mocking as documented in the project [documentation](https://webdriver.io/docs/component-testing/mocking):
+
+```ts
+import { mock, fn, unmock } from '@wdio/browser-runner'
+```
+
+To create a mock you can either create a file with the name of the module you like to mock the `__mocks__` directory, as described in [Manual Mocks](https://webdriver.io/docs/component-testing/mocking#manual-mocks), or mock the file directly as part of your test:
+
+```ts
+import { mock, fn } from '@wdio/browser-runner'
+import { format } from './utils/utils.ts'
+
+// mock files within the project
+mock('./utils/utils.ts', () => {
+    format: fn().mockReturnValue(42)
+})
```

Comment:
will this pass other exported symbols in the `./utils/utils.ts` module through? like if I had a function called `dontFormat` in there would code in this file that depended on that function still work? j/w if you need to do something like that `requireActual` thing from Jest

Reply:
If you only like to mock a subset of a module you can do it via:

```
mock('./utils/utils.ts', (origModuleFactory) => {
  const origMod = await origModuleFactory()
  return {
    ...origMod,
    format: fn().mockReturnValue(42)
  }
})
```

This is documented in [here](https://webdriver.io/docs/component-testing/mocking#partials). I thought it is too much detail for the Stencil docs but happy to add a section with docs around that here too.

Reply:
oh no need to add an explanation of that here I think, I was more just wondering. As long as we're linking to the wdio docs for this I think we're good.